cranelift-codegen/meta/src/isa/x86/legalize.rs

```diff
@@ -290,4 +297,80 @@ pub fn define(shared: &mut SharedDefinitions, x86_instructions: &InstructionGrou
+
+    let mut narrow = TransformGroupBuilder::new(
+        "x86_narrow",
+        r#"
+    Legalize instructions by narrowing.
+
+    Use x86-specific instructions if needed."#,
+    )
+    .isa("x86")
+    .chain_with(shared.transform_groups.by_name("narrow").id);
+
+    // SIMD
+    let broadcast_immediate_u8 = Literal::constant(uimm8, 0x00);
+    // let broadcast_immediate_f64 = Literal::constant(ieee64, 0x00);
+    // let lane_zero = Literal::constant(uimm8, 0x00);
+    let lane_one = Literal::constant(uimm8, 0x01);
+
+    // SIMD splat: 8-bits
+    //    narrow.legalize(
+    //        def!(y = splat.I8x16(x)), // TODO this is not possible yet
+    //        vec![
+    //            def!(a = bitcast(x)), // move into the lowest 8 bits of an XMM register TODO this encoding does not yet exist
+    //            def!(b = f64const(broadcast_immediate_f64)), // setup the shuffle mask in another XMM register; this only works if f64const zeroes out the top 64 bits TODO verify this mask is correct
+    //            def!(y = x86_pshufb(a, b)), // PSHUFB takes two XMM operands, one of which is a shuffle mask (i.e. b); TODO this encoding does not exist yet
+    //        ],
+    //    );
+
+    // SIMD splat: 16-bits
+    //    let b = var("b");
+    //    let c = var("c");
+    //    let d = var("d");
+    //    for ty in LaneTypeIterator::new().filter(|t| t.lane_bits() == 16) {
+    //        let expr = ExprBuilder::apply(splat.bind_vector(ty, 8).into(), vec![x.clone().into()])
+    //            .assign_to(vec![y.clone().into()]);
+    //        let expr_bitcast_i32x4 =
+    //            ExprBuilder::apply(bitcast.bind_vector(I32, 4).into(), vec![b.clone().into()])
+    //                .assign_to(vec![c.clone().into()]);
+    //        let expr_bitcast_x16x8 =
+    //            ExprBuilder::apply(bitcast.bind_vector(ty, 8).into(), vec![d.clone().into()])
+    //                .assign_to(vec![y.clone().into()]);
+    //
+    //        narrow.legalize(
+    //            expr, // def!(y = splat.X16x8(x)),
+    //            vec![
+    //                def!(a = x86_scalar_to_vector(x)), // translate to an x86 MOV to get the value in the lowest 16 bits of an XMM register
+    //                def!(b = insertlane(a, lane_one, x)), // insert the value again but in the next lowest 16 bits
+    //                expr_bitcast_i32x4, // def!(c = bitcast.I32#4(b)), // pretend this is an I32x4 so we can use PSHUFD, this should translate to a NOOP
+    //                def!(d = x86_pshuf(c, broadcast_immediate_u8)), // broadcast the bytes in the XMM register with PSHUF
+    //                expr_bitcast_x16x8, // def!(y = bitcast.ty#8(d)), // pretend this is an X16x8 again, this should translate to a NOOP
+    //            ],
+    //        );
+    //    }
+
+    // SIMD splat: 32-bits
+    for ty in LaneTypeIterator::new().filter(|t| t.lane_bits() == 32) {
+        narrow.legalize(
+            def!(y = splat.ty#4(x)),
```

Comment:
I'm open to suggestions but this (i.e. `T#N`) seemed like the closest way to clearly show that we are doing a SIMD binding for this instruction; Rust macros won't see `x` in `I32x4` as a separate token so we need a different syntax.

Reply:
I'm removing it; @bnjbvr's suggestion to just bind the instruction before the `def!` is more straightforward.